cypress/e2e/gui/home-page-gui.cy.js

```diff
@@ -0,0 +1,45 @@
+/// <reference types="Cypress" />
+
+describe("EngageSphere", () => {
+  beforeEach(() => {
+    cy.visit("https://engage-sphere.vercel.app/");
+  });
+
+  it("Renderiza o cabeçalho com um h1 e alternância de tema", () => {
+    cy.get("h1").contains("EngageSphere");
+    cy.get("#theme-toggle-button").should("be.visible").click();
+    cy.get("h1").contains("EngageSphere");
+  });
+  it("Exibe a saudação padrão", () => {
+    cy.get("p").eq(0).contains("Hi there!");
```

Comment:
Que tal assim?

```suggestion
    cy.contains("p", "Hi there!").should("be.visible");
```

De novo, lembre de verificar pela visibilidade do elemento, um elemento pode ser encontrado pelo Cypress no DOM e não estar visível, portanto, é importante garantir a visibilidade, se tal elemento deve estar visível.